### PR TITLE
Various formatting improvements

### PR DESCRIPTION
**Pull request by royrapoport**
Please do not review until I indicate that it's finished/ready

### CODE REVIEW

**Comment by royrapoport** [comment]
@jdq22 ready for review

File: slack_formatter.py [hunk 1]
```diff
@@ -41,15 +42,30 @@ def text_block(self, text, markdown=True):
     def divider(self):
         return { "type": "divider" }
 
-    def simple_comparison(self, cur_item, prev_item, found_prev=True, print_num=True, is_percent=False):
+    def simple_comparison(self, cur_item, prev_item, found_prev=True, print_num=True, is_percent=False, label=None):
```
Comment (jdq22) [inline slack_formatter.py:88]: This is quite the magical function.

My recommendation would be to break it into at least two smaller chunks, perhaps keeping the type casting separate from the the string manipulation at the bottom.

Will you also please add a docblock?

**Review by jdq22 — CHANGES_REQUESTED** [review]

File: slack_formatter.py (hunk 1 above)
Comment (royrapoport) [reply slack_formatter.py:88]: Thank you! Will do!

**Comment by royrapoport** [comment]
@jdq22 modularized simple_formatter -- take a look and let me know what you think

**Review by jdq22 — APPROVED** [review]
This looks fantastic. 👏

**Pull request merged**